javalib/src/main/scala/java/util/concurrent/ConcurrentSkipListSet.scala

```diff
@@ -0,0 +1,423 @@
+package java.util.concurrent
+
+import java.util._
+import java.lang.Comparable
+import scala.collection.mutable
+import scala.math.Ordering
+
+import scala.collection.JavaConversions._
+
+class ConcurrentSkipListSet[E] protected (ordering: Ordering[E], provided: Option[Comparator[E]])
+    extends AbstractSet[E]
+    with NavigableSet[E]
+    with Cloneable
+    with Serializable { self =>
+
+  def this() =
+    this(ConcurrentSkipListSet.defaultOrdering[E], None)
+
+  def this(comparator: Comparator[E])  =
+    this(Ordering.comparatorToOrdering(comparator), None)
+
+  def this(sortedSet: SortedSet[E]) = {
+    this(Ordering.comparatorToOrdering(sortedSet.comparator()).asInstanceOf[Ordering[E]],
+        Some(sortedSet.comparator().asInstanceOf[Comparator[E]]))
+    addAll(sortedSet)
+  }
+
+  def this(collection: java.util.Collection[E]) = {
+    this(ConcurrentSkipListSet.defaultOrdering[E], None)
+    addAll(collection)
+  }
+
+  def this(comparator: Comparator[E], collection: Collection[E]) = {
```

Comment:
Actually, this constructor does not appear to exist at all in the JavaDoc, does it?